packages/next/build/utils.ts

```diff
@@ -969,13 +969,17 @@ export async function buildStaticPaths({
         ) {
           paramValue = []
         }
+
         if (
           (repeat && !Array.isArray(paramValue)) ||
+          (repeat &&
+            Array.isArray(paramValue) &&
+            !paramValue.every((param) => typeof param === 'string')) ||
```

Comment:
actually you can improve a bit here, you dont need to traverse all the params, just find the first wrong one (since the error does not specify how many or which params are wrong)

change to: 
``` javascript
(repeat && Array.isArray(paramValue) &&
paramValue.some((param) => typeof param !== 'string')) || ...
```

e.g. here stops at NaN, and you avoid going further which makes no sense at this point.

``` javascript
const paramValue = ['string 1', 'string 2', NaN, 'string 3', undefined, 8]
    
if(Array.isArray(paramValue) &&
 paramValue.some((param) => typeof param !== 'string')) {
  console.log('no valid')
 } else {
  console.log('valid')
 }



```